pkg/proxy/util/utils.go

```diff
@@ -311,3 +322,54 @@ func EnsureSysctl(sysctl utilsysctl.Interface, name string, newVal int) error {
 	}
 	return nil
 }
+
+// DialContext is a dial function matching the signature of net.Dialer.DialContext.
+type DialContext = func(context.Context, string, string) (net.Conn, error)
+
+// FilteredDialOptions configures how a DialContext is wrapped by NewFilteredDialContext.
+type FilteredDialOptions struct {
+	// DialHostIPDenylist restricts hosts from being dialed.
+	DialHostCIDRDenylist []*net.IPNet
+	// DenyNonproxyableHosts restricts connections to non-proxyable hosts (as defined by
+	// IsProxyableIP).
+	DenyNonproxyableHosts bool
+}
+
+// NewFilteredDialContext returns a DialContext function that filters connections based on a FilteredDialOptions.
+func NewFilteredDialContext(wrapped DialContext, opts *FilteredDialOptions) DialContext {
```

Comment:
take a `resolv Resolver` rather than hard-coding to `net.DefaultResolver`? will allow easier unit testing

Reply:
Done